src/plugins/data/public/query/query_string/dataset_service/lib/index_type.ts

```diff
@@ -158,9 +152,22 @@ const fetchIndices = async (dataStructure: DataStructure): Promise<string[]> =>
 
   const searchResponseToArray = (response: any) => {
     const { rawResponse } = response;
-    return rawResponse.aggregations
-      ? rawResponse.aggregations.indices.buckets.map((bucket: { key: any }) => bucket.key)
-      : [];
+    if (!rawResponse.aggregations) {
+      return [];
+    }
+
+    return rawResponse.aggregations.indices.buckets.map((bucket: { key: string }) => {
+      const key = bucket.key;
+      // Handle the case of serverless cluster where key format is either:
+      // - datasource-id::TIMESERIES::<index-name>:0
+      // - datasource-id::<index-name>:0
+      // Note: Index names cannot contain ':' or '::' in OpenSearch, so these delimiters
```

Comment:
we have verified this correct?

Reply:
can we get sign off or buy in from the team formatting indices this way? it might even be worth export a const like DELIMINATER and using it. i think we have one in some plugins cant remember if it is global.

 ideally in the future collections functions like workspaces accurate results since the datasource id i believe is getting long. and then our dataset takes the id over the parent and appends itself to it. our url is getting quiet long. Discover doesn't really need to know this information if the data source already knows what information. 
 
 also might need some insight on the collection concept.